### PR TITLE
Homework 8 task 2

### PR DESCRIPTION
**Pull request by ivanzuev78**

### CODE REVIEW

File: HomeWork_8/task_2.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import sqlite3
+
+
+class TableData:
+    def __init__(self, database_name: str, table_name: str):
+        if not os.path.isfile(database_name):
+            raise FileNotFoundError("No such database")
+        self.database_name = database_name
+
+        self.conn = sqlite3.connect(database_name)
```
Comment (Cjkjvfnby) [inline HomeWork_8/task_2.py:11]: How do you close it?

File: HomeWork_8/task_2.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        self.cursor = self.conn.cursor()
+
+        if table_name not in map(
+            lambda x: x[0],
+            self.cursor.execute(
```
Comment (Cjkjvfnby) [inline HomeWork_8/task_2.py:16]: You can do a select by table name and check if row is present ifstead of reading all tables.

File: HomeWork_8/task_2.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            self.conn.close()
+            raise AttributeError(f"No such table {table_name} in database")
+
+        self.table_name = table_name
+
+    def __getitem__(self, item):
+        data = self.cursor.execute(
+            f"SELECT * from {self.table_name} where name=:name", {"name": item}
+        ).fetchone()
+        names_field = list(map(lambda x: x[0], self.cursor.description))
```
Comment (Cjkjvfnby) [inline HomeWork_8/task_2.py:30]: do you need a list here?

File: HomeWork_8/test_task_2.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+from HomeWork_8.task_2 import TableData
+
+import pytest
+
+
+@pytest.fixture()
+def database_name():
+    database_name = "test_database.sqlite"
+    conn = sqlite3.connect(database_name)
```
Comment (Cjkjvfnby) [inline HomeWork_8/test_task_2.py:12]: What is a better solution?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_8/task_2.py (hunk 3 above)
Comment (ivanzuev78) [reply HomeWork_8/task_2.py:30]: No, we don't.

File: HomeWork_8/task_2.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork_8/task_2.py:16]: Agree with you.
Done.

File: HomeWork_8/task_2.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_8/task_2.py:11]: I didn't close it.
Now I close it after every request.

File: HomeWork_8/test_task_2.py (hunk 4 above)
Comment (ivanzuev78) [reply HomeWork_8/test_task_2.py:14]: Should I mock `sqlite3.connect`?

Comment (ivanzuev78) [reply HomeWork_8/test_task_2.py:14]: Changed

File: HomeWork_8/test_task_2.py [hunk 5]
```diff
@@ -1,56 +1,52 @@
...
 
 import pytest
 
-
-@pytest.fixture()
-def copied_db_to_test_added_data():
-    shutil.copyfile("example_new.sqlite", "example_2.sqlite")
-    yield "example_2.sqlite"
-    os.remove("example_2.sqlite")
+os.chdir(os.path.dirname(__file__))
```
Comment (Cjkjvfnby) [inline HomeWork_8/test_task_2.py:8]: See comment for task 10

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**